pkg/apis/componentconfig/types.go

```diff
@@ -111,6 +111,10 @@ type KubeSchedulerConfiguration struct {
 	// Indicate the "all topologies" set for empty topologyKey when it's used for PreferredDuringScheduling pod anti-affinity.
 	// DEPRECATED: This is no longer used.
 	FailureDomains string
+
+	// When AlwaysCheckAllPredicates is set to true will check all predicates otherwise
+	// short circuit all predicates if one predicate fails.
+	AlwaysCheckAllPredicates bool
```

Comment:
BTW, an alternative implementation is to add this flag to [scheduler's policy config](https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/scheduler/api/types.go#L38), but I don't think we have any strict guidelines on where new config flags should go. So, this is optional.

Reply:
@bsalamat - +1 I'd expect this to be configurable in the policy config, as it is directly related to predicates, so might be preferable to put it there ?

Reply:
I agree with you. But the current method may be better understood by the user. This is the control of the policy rather than the policy itself.And if two places are configured, then we may need to consider the order of priority. What do you think?

Reply:
Priorities are computed anyway for any selected, ordering them might not be very useful.
As @bsalamat this isn't mandatory. @resouer @k82cn WDYT ?

Reply:
> And if two places are configured, then we may need to consider the order of priority. What do you think?

To be clear, I don't mean to add the flag to both policy config and componentconfig. I think we should remove it from componentconfig (current implementation) and add it only to policy config.

Reply:
@wgliang - misread you, as @bsalamat there should be one way to do such thing, it will just confuse users if we expose both.

Reply:
Ok, but I just have a question, how can we control the when create config from provider in [SchedulerConfig](https://github.com/kubernetes/kubernetes/blob/master/plugin/cmd/kube-scheduler/app/server.go#L657),if we just through policy config?
@bsalamat

Reply:
you might want take a look at

https://github.com/kubernetes/kubernetes/blob/7706957324c92bbb3fd4481856c5b488d04c4d94/plugin/pkg/scheduler/factory/factory.go#L847-L884

Reply:
@yastij Yes, this is create from policy config, we can add a flag in struct configFactory, but how can we control the flag if user use CreateFromProvider().

Reply:
Creating from provider should default the value to true

Reply:
> we can add a flag in struct configFactory, but how can we control the flag if user use CreateFromProvider().

You should add the flag to both configFactory struct and [the scheduler policy](https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/scheduler/api/types.go#L38). 
When a scheduler is created from "Provider", default values are used.